selftest/spinlocks/main.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import "C"
+
+import (
+	"os"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"encoding/binary"
+	"fmt"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+)
+
+func resizeMap(module *bpf.Module, name string, size uint32) error {
+	m, err := module.GetMap("events")
+	if err != nil {
+		return err
+	}
+
+	if err = m.Resize(size); err != nil {
+		return err
+	}
+
+	if actual := m.GetMaxEntries(); actual != size {
+		return fmt.Errorf("map resize failed, expected %v, actual %v", size, actual)
+	}
+
+	return nil
+}
+
+func main() {
+
+	bpfModule, err := bpf.NewModuleFromFile("main.bpf.o")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+	defer bpfModule.Close()
+
+	if err = resizeMap(bpfModule, "events", 8192); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	err = bpfModule.BPFLoadObject()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	bpfModule.ListProgramNames()
+
+	prog, err := bpfModule.GetProgram("mmap_fentry")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	link, err := prog.AttachGeneric()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+	if link.GetFd() == 0 {
+		os.Exit(-1)
+	}
+
+	eventsChannel := make(chan []byte)
+	rb, err := bpfModule.InitRingBuf("events", eventsChannel)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	lostEventCounterMap, err := bpfModule.GetMap("counter_hash_map")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	var lostEventCounterKey uint32 = 1
+	var zero uint32
+	lostEventCounterMap.Update(unsafe.Pointer(&lostEventCounterKey), unsafe.Pointer(&zero))
+
+	rb.Start()
+
+	numberOfEventsReceived := 0
+	go func() {
+		for {
+			syscall.Mmap(999, 999, 999, 1, 1)
```

Comment:
Is this even working for you ? I added an error check:

<img width="561" alt="image" src="https://user-images.githubusercontent.com/7395852/175462752-67122ec1-e728-44c9-a0e6-69fa88fbed6a.png">

and I'm getting:

<img width="643" alt="image" src="https://user-images.githubusercontent.com/7395852/175462774-00d897af-f64c-46a6-96ab-2fe68afff557.png">

and error from the syscall.

Reply:
I wonder why you picked those arguments for the mmap syscall, any particular reason ? Would an anonymous memory mmap (at a random place) be enough ?

Reply:
It's just to trigger the bpf code, the actual mmap call is irrelevant.

Reply:
Ah ok, =D makes sense then.